rmw_opendds_cpp/src/rmw_publisher.cpp

```diff
@@ -481,25 +447,25 @@ rmw_destroy_publisher(rmw_node_t * node, rmw_publisher_t * publisher)
     node_info->publisher_listener->remove_information(
       publisher_info->dds_publisher_->get_instance_handle(), EntityType::Publisher);
     node_info->publisher_listener->trigger_graph_guard_condition();
-    DDS::Publisher * dds_publisher = publisher_info->dds_publisher_;
-
-    if (dds_publisher) {
-      if (publisher_info->topic_writer_) {
-        if (dds_publisher->delete_datawriter(publisher_info->topic_writer_) != DDS::RETCODE_OK) {
-          RMW_SET_ERROR_MSG("failed to delete datawriter");
-          return RMW_RET_ERROR;
-        }
-        publisher_info->topic_writer_ = nullptr;
-      }
-      if (participant->delete_publisher(dds_publisher) != DDS::RETCODE_OK) {
-        RMW_SET_ERROR_MSG("failed to delete publisher");
-        return RMW_RET_ERROR;
-      }
-      publisher_info->dds_publisher_ = nullptr;
-    } else if (publisher_info->topic_writer_) {
-      RMW_SET_ERROR_MSG("cannot delete datawriter because the publisher is null");
-      return RMW_RET_ERROR;
-    }
+    //DDS::Publisher * dds_publisher = publisher_info->dds_publisher_;
+
+    //if (dds_publisher) {
+    //  if (publisher_info->topic_writer_) {
+    //    if (dds_publisher->delete_datawriter(publisher_info->topic_writer_) != DDS::RETCODE_OK) {
+    //      RMW_SET_ERROR_MSG("failed to delete datawriter");
+    //      return RMW_RET_ERROR;
+    //    }
+    //    publisher_info->topic_writer_ = nullptr;
+    //  }
+    //  if (participant->delete_publisher(dds_publisher) != DDS::RETCODE_OK) {
+    //    RMW_SET_ERROR_MSG("failed to delete publisher");
+    //    return RMW_RET_ERROR;
+    //  }
+    //  publisher_info->dds_publisher_ = nullptr;
+    //} else if (publisher_info->topic_writer_) {
+    //  RMW_SET_ERROR_MSG("cannot delete datawriter because the publisher is null");
+    //  return RMW_RET_ERROR;
+    //}
```

Comment:
Is this code that isn't compiling but needs to be revisited or is it something we need to get rid of?

Reply:
This code is not needed after pointers have been replaced with smart pointers.